examples/flex.rs

```diff
@@ -1,326 +1,452 @@
-        let areas = &Layout::vertical([Fixed(EXAMPLE_HEIGHT); N_EXAMPLES_PER_TAB as usize])
-            .flex(Flex::Start)
-            .split(area);
-
-        let examples = [
-            vec![Length(20), Fixed(20), Percentage(20)],
-            vec![Fixed(20), Percentage(20), Length(20)],
-            vec![Percentage(20), Length(20), Fixed(20)],
-            vec![Length(20), Length(15)],
-            vec![Length(20), Fixed(20)],
-            vec![Min(20), Max(20)],
-            vec![Max(20)],
-            vec![Min(20), Max(20), Length(20), Fixed(20)],
-            vec![Proportional(0), Proportional(0)],
-            vec![Proportional(1), Proportional(1), Length(20), Fixed(20)],
-            vec![
-                Min(10),
-                Proportional(3),
-                Proportional(2),
-                Length(15),
-                Fixed(15),
-            ],
-        ];
-
-        for (area, constraints) in areas.iter().zip(examples) {
-            Example::new(constraints).flex(flex).render(*area, buf);
+impl SelectedTab {
+    fn render_examples(&self, area: Rect, buf: &mut Buffer, flex: Flex) {
+        let heights = EXAMPLE_DATA
+            .iter()
+            .map(|(desc, _)| get_description_height(desc) + 4);
+        let areas = Layout::vertical(heights).flex(Flex::Start).split(area);
+        for (area, (description, constraints)) in areas.iter().zip(EXAMPLE_DATA.iter()) {
+            Example::new(constraints, description, flex).render(*area, buf);
         }
     }
 }
 
-struct Example {
-    constraints: Vec<Constraint>,
-    flex: Flex,
-}
-
 impl Example {
-    fn new<C>(constraints: C) -> Self
-    where
-        C: Into<Vec<Constraint>>,
-    {
+    fn new(constraints: &[Constraint], description: &str, flex: Flex) -> Self {
         Self {
             constraints: constraints.into(),
-            flex: Flex::default(),
+            description: description.into(),
+            flex,
         }
     }
-
-    fn flex(mut self, flex: Flex) -> Self {
-        self.flex = flex;
-        self
-    }
 }
 
 impl Widget for Example {
     fn render(self, area: Rect, buf: &mut Buffer) {
-        let [legend, area] = area.split(&Layout::vertical([Ratio(1, 3); 2]));
+        let title_height = get_description_height(&self.description);
+        let layout = Layout::vertical([Fixed(title_height), Proportional(0)]);
+        let [title, illustrations] = area.split(&layout);
         let blocks = Layout::horizontal(&self.constraints)
             .flex(self.flex)
-            .split(area);
-
-        self.legend(legend.width as usize).render(legend, buf);
+            .split(illustrations);
+
+        if !self.description.is_empty() {
+            Paragraph::new(
+                self.description
+                    .split('\n')
+                    .map(|s| {
+                        format!("// {}", s)
+                            .italic()
+                            .fg(Color::from_str("#908caa").unwrap())
+                    })
```

Comment:
What about:
```suggestion
                    .map(|s| format!("// {}", s).italic().fg(tailwind::SLATE.c400))
```
which gives:
<img width="804" alt="image" src="https://github.com/ratatui-org/ratatui/assets/381361/55bed1c1-33a4-4445-b335-cdfc052fc51a">

instead of:

<img width="785" alt="image" src="https://github.com/ratatui-org/ratatui/assets/381361/117994df-f1fb-4bd7-b221-cb5733cd7ea8">

Reply:
It could be worth also manually setting the background color of the app if we're drawing on that directly.
I just realized this probably looks fine on a black bg (or something with less color in it than my blue terminal)